go/kbfs/simplefs/archive.go

```diff
@@ -829,7 +858,18 @@ func (m *archiveManager) doZipping(ctx context.Context, jobID string) (err error
 
 		err = zipWriterAddDir(ctx, zipWriter, workspaceDir, updateBytesZipped)
 		if err != nil {
-			return fmt.Errorf("zipWriter.AddFS to %s error: %v", jobDesc.ZipFilePath, err)
+			return fmt.Errorf("zipWriterAddDir into %s error: %v", jobDesc.ZipFilePath, err)
+		}
+
+		{ // write the manifest
+			w, err := zipWriter.Create("manifest.json")
```

Comment:
Not sure how this works, but do you need to close `w`?

Reply:
Yeah this is a bit odd. This gives an io.Writer not an io.WriteCloser. I think the zip writer close is what flushes.